nvm.sh

```diff
@@ -2724,26 +2727,25 @@ nvm() {
     return
   fi
 
-  local DEFAULT_IFS
-  DEFAULT_IFS=" $(nvm_echo t | command tr t \\t)
+  local IFS
+  IFS=" $(nvm_echo t | command tr t \\t)
 "
-  if [ "${-#*e}" != "$-" ]; then
-    set +e
-    local EXIT_CODE
-    IFS="${DEFAULT_IFS}" nvm "$@"
-    EXIT_CODE="$?"
-    set -e
-    return "$EXIT_CODE"
-  elif [ "${-#*a}" != "$-" ]; then
-    set +a
+  local OPTIONS=''
+  local O='e a'
+  # set +E only disables errtrace on Bash
+  [ -z "${BASH-}" ] || O='e a E'
```

Comment:
Let’s use an if here instead of abusing logical operators for flow control :-)

Reply:
It's no longer applicable anyway, but I did check this pattern was used elsewhere in nvm.sh before using it 😛

Reply:
i should clean those up :-p